lib/model/folder_sendrecv.go

```diff
@@ -454,7 +454,7 @@ nextFile:
 
 		// Check our list of files to be removed for a match, in which case
 		// we can just do a rename instead.
-		key := string(fi.Blocks[0].Hash)
+		key := string(fi.BlocksHash)
 		for i, candidate := range buckets[key] {
 			if candidate.BlocksEqual(fi) {
```

Comment:
This `if` just checks the same thing we used as the key (the `BlocksHash`) so we could drop it

Reply:
Even when trying to match a delete, we still check block by block. I don't think there is that much harm in checking this.

Reply:
BlocksEqual just checks the hash if there is one, but yeah it doesn't do any harm as is.

Reply:
Ah yes! I thought it still compared the blocks, I'll fix that.